src/index.js

```diff
@@ -52,12 +52,12 @@ export default function createStore(state) {
 				for (let i=0; i<arguments.length; i++) args.push(arguments[i]);
 				let ret = action.apply(this, args);
 				if (ret!=null) {
-					if (ret.then) ret.then(setState);
-					else setState(ret);
+					if (ret.then) ret.then(res => setState(res, false, action));
```

Comment:
Idea: we could hoist the action -> setState() proxy to L49:

```js
action(action) {
  function apply(result) {
    setState(result, false, action);
  }
  return function() {
    // ... etc
    if (ret.then) ret.then(apply);
    else apply(ret)
  };
}
```

That should save a few bytes, and it avoids the nested closure + return!